magento/magento2
# Catalog ProductList helper has coupling to Registry #27119

### Description (*)
In this pull request the **Magento\Framework\Registry** class dependency is removed from the **Magento\Catalog\Helper\Product\ProductList** class.

The new **Magento\Catalog\Model\Category\Toolbar\Config** class has been added to provide access to the **Sort By** (Order Field) value. 

The **getOrderField()** method checks if the current category has the **default_sort_by** value, and returns it, the default configuration value is returned otherwise.

### Fixed Issues (if relevant)
1. Fixes magento/magento2#27119: Catalog ProductList helper has coupling to Registry

### Manual testing scenarios (*)
1. Create a Category and add two Simple Products into the Category
2. Clear Cache
3. Verify that the Category opens correctly and uses the default **Sort By: Position** setting

### Contribution checklist (*)
